src/puya/ir/optimize/collapse_blocks.py

```diff
@@ -119,7 +119,7 @@ def remove_empty_blocks(_context: CompileContext, subroutine: models.Subroutine)
             empty_block = block
             target = block.terminator.target
 
-            if not empty_block.predecessors and block is not subroutine.body[0]:
+            if not empty_block.predecessors and block is subroutine.body[0]:
```

Comment:
the condition was correct before, if a bit confusing.
unreachable empty blocks were deferred for removal until the `remove_unreachable_blocks` pass, except for the entry block which is always considered unreachable.
we can actually just delete this `if/continue` block entirely.

Reply:
I see you found the actual cause of the bug and fixed it. I'll rebase and revert this change